experimental/oak_baremetal_loader/src/main.rs

```diff
@@ -186,60 +211,64 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let comms = vmm.create_comms_channel().await?;
 
     // A message based communication channel that permits other parts of the
-    // untrusted loader to send requests to the tasks that handles communicating
-    // with the runtime and receive responses. Requests and (ok) responses
-    // sent over this channel represent flatbuffer encoded messages. This is
-    // done as it is not possible to send the flatbuffer Message structs
-    // themselves over the channel.
+    // untrusted loader to send requests to the task that handles communicating
+    // with the runtime and receive responses.
     let (request_dispatcher, mut request_receiver) =
-        bmrng::unbounded_channel::<Vec<u8>, Result<Vec<u8>, oak_idl::Status>>();
+        bmrng::unbounded_channel::<oak_idl::Request, Result<Vec<u8>, oak_idl::Status>>();
```

Comment:
Note that we could alternatively also replace this channel with a central client, that is shared to different parts of the untrusted loader via mutex. The result would be pretty much the same. 

The bmrng channel (and more generally sharing state via messages) has kind of grown on me though, so I'm happy to keep it.